distributed/nanny.py

```diff
@@ -848,10 +848,15 @@ def watch_stop_q():
                 Wait for an incoming stop message and then stop the
                 worker cleanly.
                 """
-                msg = child_stop_q.get()
-                child_stop_q.close()
-                assert msg.pop("op") == "stop"
-                loop.add_callback(do_stop, **msg)
+                try:
+                    msg = child_stop_q.get()
+                except (TypeError, OSError):
```

Comment:
I'm uncomfortable catching and hiding the `TypeError` without knowing why this might occur. Could you describe the situations in which it might happen? Also, it's probably worth catching `ValueError` too as it looks like `OSError` was thrown prior to Python 3.8.

> get([block[, timeout]])
>    Remove and return an item from the queue. If optional args block is True (the default) and timeout is None (the default), block if necessary until an item is available. If timeout is a positive number, it blocks at most timeout seconds and raises the [queue.Empty](https://docs.python.org/3/library/queue.html#queue.Empty) exception if no item was available within that time. Otherwise (block is False), return an item if one is immediately available, else raise the queue.Empty exception (timeout is ignored in that case).
>   Changed in version 3.8: If the queue is closed, ValueError is raised instead of OSError.